code/modules/mob/living/carbon/alien/humanoid/queen.dm

```diff
@@ -10,9 +10,74 @@
 	layer = LARGE_MOB_LAYER //above most mobs, but below speechbubbles
 	pressure_resistance = 200 //Because big, stompy xenos should not be blown around like paper.
 	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab/xeno = 20, /obj/item/stack/sheet/animalhide/xeno = 3)
+	gps_name = "Hissing Signal"
 
+	var/life_ticks_to_wait = 10 //life does some expensive things, so we only want to do the calculation occasionally
 	var/alt_inhands_file = 'icons/mob/alienqueen.dmi'
 
+/mob/living/carbon/alien/humanoid/royal/queen/Login()
+	..()
+	SSshuttle.registerHostileEnvironment(src)
+
+/mob/living/carbon/alien/humanoid/royal/queen/Logout()
+	..()
+	addtimer(CALLBACK(src, .proc/logoutcheck), 12 SECONDS)//when logging out, we don't want to spam announcements with leaving and reconnecting.
+	SSshuttle.clearHostileEnvironment(src)
+	neutralized = TRUE
+
+/mob/living/carbon/alien/humanoid/royal/queen/death()
+	..()
+	SSshuttle.clearHostileEnvironment(src)
+	neutralized = TRUE
+
+/mob/living/carbon/alien/humanoid/royal/queen/proc/logoutcheck()
+	if(!client)
+		SSshuttle.clearHostileEnvironment(src) //the queen has officially gone braindead, so we can allow the shuttle to leave
+
+/mob/living/carbon/alien/humanoid/royal/queen/Life()

```

Comment:
Also this entire check has no business being on the queen either.

Reply:
Thanks a ton for the input. I do appreciate it. I've actually been tinkering off and on with this PR and getting a bit of help from coderbus for awhile. I'll go ahead and work on moving the things you mentioned.